### PR TITLE
Adding Filters to MonitorChartPart

### PR DESCRIPTION
**Pull request by Orkan-M**
Pushed as draft as unable to test other assertion tests as shown in image below. Because it's .NET 5, all the below errors are shown when cloning master. Not sure how to fix without mass-supressing them.

image

Otherwise, this PR adds functionality to add custom filters into the JSON schema which should allow users to compares values, such as `"EntityName" "=" "test1, "test2"`. This would create a graph that has test1 and test2 shown.

### CODE REVIEW

File: src/Kustomaur.Models/Filters/FilterModel.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Kustomaur.Models.Filters
+{
+    public class FilterModel
+    {
+        public string Name { get; set; }
+        public string Operator { get; set; }
```
Comment (jmahmud) [inline src/Kustomaur.Models/Filters/FilterModel.cs]: can we make this an enumeration in order to fix which operators can be used?

File: src/Kustomaur.Builder/DashboardParts/Implementations/MonitorChartPart.cs [hunk 2]
```diff
@@ -103,7 +112,20 @@ private Input BuildChartInput()
...
-            
+
+            if (_filters != null && _filters.Any())
+            {
+                value.Chart.Options = new ChartOptions
+                {
+                    Filters = _filters.Select(filter => new FilterModel
+                    {
+                        Name = filter.Key,
+                        Operator = "equals",
```
Comment (jmahmud) [inline src/Kustomaur.Builder/DashboardParts/Implementations/MonitorChartPart.cs]: I don't think Operator should defaultto a string like this....

File: tests/Kustomaur.Builder.Tests/DashboardFilters/FilterModelTests.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+using System.Threading.Tasks;
+using Kustomaur.Models.Filters;
+using Xunit;
+
+namespace Kustomaur.Builder.Tests.DashboardFilters
+{
+    public class FilterModelTests
+    {
+        [Fact]
+        public void FilterModel_PropertiesAreSetCorrectly()
```
Comment (jmahmud) [inline tests/Kustomaur.Builder.Tests/DashboardFilters/FilterModelTests.cs:12]: Can we also add a test which asserts the JSON that is produced?

**Review by jmahmud — CHANGES_REQUESTED** [review]

File: tests/Kustomaur.Builder.Tests/DashboardFilters/FilterModelTests.cs (hunk 3 above)
Comment (Orkan-M) [reply tests/Kustomaur.Builder.Tests/DashboardFilters/FilterModelTests.cs:12]: Edited.

**Comment by Orkssss** [comment]
Ready for the latest review :)

**Comment by Orkan-M** [comment]
Still having formatting issues when the JSON gets generated. Enum toString is being used too, which isn't ideal.

image

It's populating values incorrectly.

**Comment by jmahmud** [comment]
> Still having formatting issues when the JSON gets generated. Enum toString is being used too, which isn't ideal.
> 
> image
> 
> It's populating values incorrectly.

**Comment by jmahmud** [comment]
@Orkssss - although not very TDD, the test does a  blanket JSON validation - therefore as you have altered the datastructre of what to be rendered, you'll need to update the test....

**Pull request merged**